From 2056c6802c5afbd722fcffc99a5cf50bbaee23c8 Mon Sep 17 00:00:00 2001
From: Tom Rini <trini@ti.com>
Date: Tue, 15 Nov 2011 17:43:31 -0700
Subject: [PATCH 1/3] omap3:am3517evm: Allow for NAND to really be disabled

On boards with both NAND and NOR and NOR being selected, NAND needs
to be disabled fully.

Signed-off-by: Tom Rini <trini@ti.com>
---
 .../arch/arm/mach-omap2/board-am3517evm.c          |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/board-am3517evm.c b/arch/arm/mach-omap2/board-am3517evm.c
index 3992b86..1d17415 100644
--- a/arch/arm/mach-omap2/board-am3517evm.c
+++ b/arch/arm/mach-omap2/board-am3517evm.c
@@ -59,6 +59,7 @@
 
 #define NAND_BLOCK_SIZE        SZ_128K
 
+#ifdef CONFIG_MTD_NAND_OMAP2
 static struct mtd_partition am3517evm_nand_partitions[] = {
 /* All the partition sizes are listed in terms of NAND block size */
 	{
@@ -89,6 +90,7 @@ static struct mtd_partition am3517evm_nand_partitions[] = {
 		.offset         = MTDPART_OFS_APPEND,
 	},
 };
+#endif
 
 static struct mdio_platform_data am3517_evm_mdio_pdata = {
 	.bus_freq	= AM35XX_EVM_MDIO_FREQUENCY,
@@ -926,9 +928,11 @@ static void __init am3517_evm_init(void)
 	/* DSS */
 	am3517_evm_display_init();
 
+#ifdef CONFIG_MTD_NAND_OMAP2
 	/* NAND */
 	board_nand_init(am3517evm_nand_partitions,
 					ARRAY_SIZE(am3517evm_nand_partitions), 0, NAND_BUSWIDTH_16);
+#endif
 
 	/* RTC - S35390A */
 	am3517_evm_rtc_init();
-- 
1.7.0.4

